src/EscapeCodeGenerator.hpp

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include "Defs.hpp"
+#include <string>
+
+struct EscapeCodeGenerator {
+
+    using EscapeCode = std::string;
+
+    virtual EscapeCode cursorForward(int) { return EscapeCode{}; }
+    virtual EscapeCode cursorBackward(int) { return EscapeCode{}; }
+    virtual EscapeCode clearTerminalLine() { return EscapeCode{}; }
+    virtual EscapeCode setColor(Color) { return EscapeCode{}; }
+
+};
+
+struct ANSIEscapeCodeGenerator : public EscapeCodeGenerator {
+
+    EscapeCode cursorForward(int n) override {
+        if (!n) return EscapeCode{};
+        EscapeCode a{"\033["};
+        a += std::to_string(n);
+        a += 'C';
+        return a;
```

Comment:
If EscapeCode is string and always starts with this `\033[` stuff, then why these functions couldn't be like `return EscapeCode{} + std::to_string(n) + 'C'`?